CONTRIBUTING.md

```diff
@@ -73,7 +75,102 @@ would like to work on an issue, please follow these steps:
 By following this process, we can avoid duplication of efforts and ensure clear
 communication among all contributors.
 
-### Creating a New Issue
+## Rust Style Guide
+
+### Formatting
+
+- Use rustfmt with nightly features to maintain consistent code formatting.
+- Sort Cargo.toml dependencies alphabetically.
+
+### Module Organization
+
+- We don't use the `mod.rs` pattern.
+- Instead, export modules from files named according to their context.
+
+For example:
+
+```bash
+crates/meroctl/src/cli/app.rs
+```
+
+Contains:
+
+```rust
+mod get;
+mod install;
+mod list;
+```
+
+And we would have these individual files:
+
+```bash
+crates/meroctl/src/cli/app/get.rs
+crates/meroctl/src/cli/app/install.rs
+crates/meroctl/src/cli/app/list.rs
+```
+
+### Error Handling
+
+- Almost no unwrapping (acceptable in tests and possibly when dealing with thread join handlers).
+- If unwrapping is absolutely necessary, explain why with a comment.
+- On values that may return errors, use `.map_err()` to map the error into the appropriate Error type used in that crate/module.
+
+### Code Efficiency
+
+- Try to limit unnecessary clones.
+- Use short-circuiting if statements:
+
+```rust
+// NOT RECOMMENDED:
+if some_condition {
+    // ... (lots of code)
+}
+
+// RECOMMENDED:
+if !some_condition {
+    return Err(YourError::Something);
+}
+// Continue with main code path...
+```
```

Comment:
it's only an issue when there's an `else` case

````suggestion
```rust
// NOT RECOMMENDED:
if some_condition {
    // ... (lots of code)
} else {
    return Err(YourError::Something);
}

// RECOMMENDED:
if !some_condition {
    return Err(YourError::Something);
}
// Continue with main code path...
```
````